great_expectations/datasource/data_connector/inferred_asset_azure_data_connector.py

```diff
@@ -0,0 +1,132 @@
+
+    The InferredAssetAzureDataConnector is one of two classes (ConfiguredAssetAzureDataConnector being the
+    other one) designed for connecting to filesystem-like data, more specifically files on Azure blob store. It
+    connects to assets inferred from container, prefix, and file name by default_regex.
+
+    As much of the interaction with the SDK is done through a BlobServiceClient, please refer to the official
+    docs if a greater understanding of the supported authentication methods and general functionality is desired.
+    Source: https://docs.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.blobserviceclient?view=azure-python
+    """
+
+    def __init__(
+        self,
+        name: str,
+        datasource_name: str,
+        container: str,
+        execution_engine: Optional[ExecutionEngine] = None,
+        default_regex: Optional[dict] = None,
+        sorters: Optional[list] = None,
+        name_starts_with: str = "",
+        delimiter: str = "/",
+        azure_options: Optional[dict] = None,
+        batch_spec_passthrough: Optional[dict] = None,
+    ):
+        """
+        InferredAssetAzureDataConnector for connecting to Azure Blob store.
+
+        Args:
+            name (str): required name for data_connector
+            datasource_name (str): required name for datasource
+            container (str): container for Azure blob store
+            execution_engine (ExecutionEngine): optional reference to ExecutionEngine
+            default_regex (dict): optional regex configuration for filtering data_references
+            sorters (list): optional list of sorters for sorting data_references
+            name_starts_with (str): Azure prefix
+            delimiter (str): Azure delimiter
+            azure_options (dict): wrapper object for **kwargs
+            batch_spec_passthrough (dict): dictionary with keys that will be added directly to batch_spec
+        """
+        logger.debug(f'Constructing InferredAssetAzureDataConnector "{name}".')
+
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            execution_engine=execution_engine,
+            default_regex=default_regex,
+            sorters=sorters,
+            batch_spec_passthrough=batch_spec_passthrough,
+        )
+
+        self._container = container
+        self._name_starts_with = os.path.join(name_starts_with, "")
+        self._delimiter = delimiter
+
+        if azure_options is None:
+            azure_options = {}
+
+        try:
+            if "conn_str" in azure_options:
+                self._azure = BlobServiceClient.from_connection_string(**azure_options)
+            else:
+                self._azure = BlobServiceClient(**azure_options)
+        except (TypeError, AttributeError):
+            raise ImportError(
+                "Unable to load Azure BlobServiceClient (it is required for InferredAssetAzureDataConnector)."
+            )
```

Comment:
Please follow the `try/except` logic from the ConfiguredAssetAzureDataConnector; we require regex to parse our `self._account_name` (which is used in `_get_full_file_path()`)